contracts/adapters/FlexibleLeverageStrategyAdapter.sol

```diff
@@ -469,6 +554,36 @@ contract FlexibleLeverageStrategyAdapter is BaseAdapter {
         return _calculateCurrentLeverageRatio(currentLeverageInfo.collateralValue, currentLeverageInfo.borrowValue);
     }
 
+    /**
+     * Calculates the total notional rebalance size.
+     *
+     * @return uint256      total notional rebalance size
+     */
+    function getTotalRebalanceNotional()  external view returns(uint256) {
+
+        uint256 newLeverageRatio;
+        uint256 currentLeverageRatio = getCurrentLeverageRatio();
+
+        if (currentLeverageRatio > incentive.incentivizedLeverageRatio) {
```

Comment:
I can guess why you're breaking it out this way instead of using `_calculateNewLeverageRatio`, but we need some clarifying comments on why and lay out the scenarios we're trying to cover here. I.e. there could be situations where we're in ripcord territory, but there is a twapLeverageRatio set from a previous iterateRebalance, which would overridden by a ripcord call. Therefore, we want to check if it's above the incentivized leverage ratio directly